test/test_cpp_extensions_aot.py

```diff
@@ -164,6 +168,48 @@ def test_cuda_dlink_libs(self):
         test = cuda_dlink.add(a, b)
         self.assertEqual(test, ref)
 
+    @unittest.skipIf(not TEST_CUDA, "python_agnostic is a CUDA extension + needs CUDA")
+    @unittest.skipIf(not common.IS_LINUX, "test requires linux tools ldd and nm")
+    def test_python_agnostic(self):
+        # For this test, run_test.py will call `python setup.py bdist_wheel` in the
+        # cpp_extensions/python_agnostic_extension folder, where the extension and
+        # setup calls specify py_limited_api to `True`. To approximate that the
+        # extension is indeed python agnostic, we test
+        #   a. The extension wheel name contains "cp39-abi3", meaning the wheel
+        # should be runnable for any Python 3 version after and including 3.9
+        #   b. The produced shared library does not have libtorch_python.so as a
+        # dependency from the output of "ldd _C.so"
+        #   c. The .so does not need any python related symbols. We approximate
+        # this by running "nm -u _C.so" and grepping that nothing starts with "Py"
+
+        dist_root = os.path.join("cpp_extensions", "python_agnostic_extension", "dist")
+        matches = list(Path(dist_root).glob("*.whl"))
+        self.assertEqual(len(matches), 1, msg=str(matches))
+        whl_file = matches[0]
+        self.assertRegex(str(whl_file), r".*python_agnostic-0\.0-cp39-abi3-.*\.whl")
```

Comment:
I'm confused: why do you require minimum CPython version 3.9? You're not using /any/ CPython APIs right?

Reply:
Because that's the minimum version we support?

Reply:
Yea, I think Ed's confusion is of the same essence as what you had asked yesterday, so let me answer here for posterity:

There is indeed a difference between the test case here and the API that is offered, namely:
(1) the custom extension here (or torchao haha) does NOT use any CPython APIs, and is therefore Python-free
but (2) the py-limited-api flag that CPython/Setup/PyPI respects allows for limited CPython use, and only signifies Python-agnosticism.

Our end goal is allowing custom extensions to build 1 python wheel that should work with multiple versions of Python, which, without PyTorch, Python-related-entities already have an API for! To build a wheel that is python agnostic today, one must pass `py_limited_api` to both the underlying [Extension](https://github.com/pypa/setuptools/blob/9692cde009af4651819d18a1e839d3b6e3fcd77d/setuptools/extension.py#L127) class and the [bdist_wheel command](https://github.com/pypa/setuptools/issues/4741). 

This workflow that works for most normal Extensions just doesn't work for PyTorch Extensions because we link torch_python no matter what :') SO the change we're making on our end is not linking torch_python when the flag is set. I will add some documentation making this exclusion explicit to users. (This workflow is terribly documented and there are not even 500 uses on public [GitHub](https://github.com/search?q=%2F%28%3F-i%29%22py_limited_api%22%2F&type=code), so I'm not surprised no one's yelled at us yet.)

After this PR, I do plan on writing a small tutorial showing how someone could do this.